### PR TITLE
Add conj method for vectorarrays

### PR DESCRIPTION
**Pull request by pmli**
See #445.

### CODE REVIEW

File: src/pymor/vectorarrays/numpy.py [hunk 1]
```diff
@@ -48,6 +48,11 @@ def real(self):
     def imag(self):
         return NumpyVectorArray(self.data.imag.copy(), self.space)
 
+    def conj(self):
+        if np.isrealobj(self.data):
+            return self.copy()
+        return NumpyVectorArray(self.data.real.copy() - 1j * self.data.imag.copy(), self.space)
```
Comment (sdrave) [inline src/pymor/vectorarrays/numpy.py]: Any reason for not using `np.conj`?  Also, we should ensure that for a non-complex array no unneccessary copy is made. (We can simply return `self.copy()` due to COW semantics of our VectorArrays.)

**Comment by pmli** [comment]
> We will also need a default impl in `VectorArrayInterface` which just returns `self.copy()`.

@sdrave Should there also be default implementations of `real` and `imag`?

**Comment by sdrave** [comment]
Would not hurt to have default implementations for `real` and `imag`. `conj` is more important though, since we want to use it in `VectorArrayOperator`.

**Comment by pmli** [comment]
> Would not hurt to have default implementations for `real` and `imag`.

In this PR or a different one?

**Comment by pmli** [comment]
Here it is.

**Pull request merged**